environment-creator-application/src/main/java/com/ca/apim/gateway/cagatewayconfig/KeystoreCreator.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig;
+
+import com.ca.apim.gateway.cagatewayconfig.util.file.FileUtils;
+import com.ca.apim.gateway.cagatewayconfig.util.injection.ConfigBuilderModule;
+import com.ca.apim.gateway.cagatewayconfig.util.keystore.KeyStoreCreationException;
+import com.ca.apim.gateway.cagatewayconfig.util.keystore.KeystoreHelper;
+import org.apache.commons.io.IOUtils;
+
+import java.io.File;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.ca.apim.gateway.cagatewayconfig.util.file.FileUtils.closeQuietly;
+import static org.apache.commons.lang.StringUtils.isEmpty;
+
+class KeystoreCreator {
+
+    private static FileUtils fileUtils = FileUtils.INSTANCE;
+
+    private KeystoreCreator() {
+    }
+
+    static void createKeyStoreIfNecessary(Map<String, String> environmentProperties, String keyStoreDirPath) {
+        // write keystore if directory is specified and exists
+        String privateKeyFilesDirectory = environmentProperties.get(KeystoreHelper.ENV_VAR_KEYSTORE_PATH);
+        if (isEmpty(privateKeyFilesDirectory) || !Paths.get(privateKeyFilesDirectory).toFile().exists()) {
+            return;
+        }
+
+        KeystoreHelper keystoreHelper = ConfigBuilderModule.getInjector().getInstance(KeystoreHelper.class);
+        final byte[] keyStore = keystoreHelper.createKeyStoreFromEnvironment(privateKeyFilesDirectory, environmentProperties);
+        // no keys
+        if (keyStore.length == 0) {
+            return;
+        }
+
+        final File keyStoreDirectory = new File(keyStoreDirPath);
+        if (!keyStoreDirectory.exists() && !keyStoreDirectory.mkdirs()) {
+            throw new KeyStoreCreationException("Could not create directory '" + keyStoreDirPath + "' for KeyStore");
+        }
+
+        File keyStoreFile = new File(keyStoreDirectory, "keystore.gwks");
+        OutputStream stream = fileUtils.getOutputStream(keyStoreFile);
+        try {
+            IOUtils.write(keyStore, stream);
+        } catch (IOException e) {
+            throw new KeyStoreCreationException("Unexpected error writing key store", e);
+        } finally {
+            closeQuietly(stream);
+        }
+
+        // Properties with the keystore configuration
+        // FIXME this is not being used yet
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1134))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=76&issues=AWZpMB1RksgJibV52Hko&open=AWZpMB1RksgJibV52Hko)

[ISSUE_KEY]: AWZpMB1RksgJibV52Hko